src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp

```diff
@@ -174,32 +232,72 @@ struct boss_jeklik : public BossAI
             events.ScheduleEvent(EVENT_MIND_FLAY, 10s, 30s, PHASE_TWO);
             events.ScheduleEvent(EVENT_GREATER_HEAL, 25s, PHASE_TWO);
             events.ScheduleEvent(EVENT_SPAWN_FLYING_BATS, 10s, PHASE_TWO);
+        });
+    }
 
-            return;
+    void EnterEvadeMode(EvadeReason why) override
+    {
+        if (why != EvadeReason::EVADE_REASON_NO_PATH)
+        {
+            // make invisible to hide wonky-looking movement
+            me->SetVisible(false);
+
+            // cancel any pending moves and stop moving
+            me->GetMotionMaster()->Clear();
+            me->AddUnitState(UNIT_STATE_ROOT);
+
+            // cancel all pending events
+            events.CancelEventGroup(PHASE_ONE);
+            events.CancelEventGroup(PHASE_TWO);
+            scheduler.CancelAll();
         }
+
+        // enter evade mode
+        CreatureAI::_EnterEvadeMode(why);
+        Reset();
+    }
+
+    void JustDied(Unit* killer) override
+    {
+        BossAI::JustDied(killer);
+        Talk(SAY_DEATH);
     }
 
     void UpdateAI(uint32 diff) override
     {
+        events.Update(diff);
+        scheduler.Update(diff);
+
+        // update boss target (victim)
+        // if none, return
         if (!UpdateVictim())
+        {
             return;
+        }
 
-        events.Update(diff);
-
+        // if casting, return
         if (me->HasUnitState(UNIT_STATE_CASTING))
+        {
             return;
+        }
 
         while (uint32 eventId = events.ExecuteEvent())
         {
             switch (eventId)
             {
-                // Phase one
+                // PHASE_ONE
                 case EVENT_CHARGE_JEKLIK:
-                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
+                    // charge the nearest player that is at least 8 yards away (charge min distance)
```

Comment:
I'll be happy to change this, but I'd like to understand why this case needs a scheduler and the rest is okay being event-based? My understanding from asking on Discord is that either approach is acceptable, and since the rest of the AI is done using events I stuck to that.

Reply:
Yes. But I don't think seeing both in 1 script makes a lot of sense. So it's best to pick one. I would go with the scheduler, personally, as you're essentially revamping most of the script. Taskscheduler is the new standard for boss scripts

Reply:
Done, events are no more.